### PR TITLE
BillingClient 5 Update

### PR DESCRIPTION
**Pull request by beylmk**
This is the first PR of the BillingClient 5 changes. It updates the version and fixes breaking changes, but does not yet take advantage of any of the new APIs.

Note that I'm going to keep BC5 stuff on the billing-client-5 branch for now.

### CODE REVIEW

File: api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesAPI.java [hunk 1]
```diff
@@ -159,4 +159,12 @@ static void checkLogHandler() {
         );
         final LogHandler handler = Purchases.getLogHandler();
     }
+
+    static void check(final BillingFeature feature) {
```
Comment (beylmk) [inline api-tester/src/main/java/com/revenuecat/apitester/java/PurchasesAPI.java:163]: these were missing

File: build.gradle [hunk 2]
```diff
@@ -3,7 +3,7 @@ buildscript {
     ext.compileVersion = 31
     ext.buildToolsVersion = "30.0.3"
     ext.minVersion = 14
-    ext.billingVersion = "4.1.0"
+    ext.billingVersion = "5.0.0"
```
Comment (beylmk) [inline build.gradle:6]: i think @tonidero was working on splitting out a unityIAP version?

File: feature/google/src/test/java/BillingWrapperTest.kt [hunk 3]
```diff
@@ -110,7 +110,7 @@ class BillingWrapperTest {
         val billingClientPurchaseHistoryListenerSlot = slot<PurchaseHistoryResponseListener>()
         every {
             mockClient.queryPurchaseHistoryAsync(
-                any(),
+                any<String>(),
```
Comment (beylmk) [inline feature/google/src/test/java/BillingWrapperTest.kt:113]: this became ambiguous, as the new version of this function now takes `QueryPurchaseHistoryParams`

**Review by tonidero — APPROVED** [review]
I was thinking, since this is likely going to be a long term project, with a lot of opportunities for conflicts, that it would be good to merge the progress to `main` as soon as possible. I wonder if we should move to have 3 variants here... `unityIAP`, `latestDependencies` and `bc5support` or something like that. Once the work is done, we can just convert the `bc5support` flavor into `latestDependencies`. This way, we also make sure we keep compatibility with bc4 for unity. Any thoughts on this approach? 

For now, these changes make sense to me so we can go ahead with this PR and discuss this in Slack.

**Pull request merged**